src/mpid/ch4/src/ch4r_win.h

```diff
@@ -83,7 +83,25 @@ static inline int MPIDI_CH4R_mpi_win_set_info(MPIR_Win * win, MPIR_Info * info)
             if (!strcmp(curr_ptr->value, "same_op"))
                 MPIDI_CH4U_WIN(win, info_args).accumulate_ops = MPIDI_CH4I_ACCU_SAME_OP;
         }
-    next:
+        else if (!strcmp(curr_ptr->key, "same_size")) {
+            if (!strcmp(curr_ptr->value, "true"))
+                MPIDI_CH4U_WIN(win, info_args).same_size = 1;
+            else
+                MPIDI_CH4U_WIN(win, info_args).same_size = 0;
+        }
+        else if (!strcmp(curr_ptr->key, "same_disp_unit")) {
+            if (!strcmp(curr_ptr->value, "true"))
+                MPIDI_CH4U_WIN(win, info_args).same_disp_unit = 1;
+            else
+                MPIDI_CH4U_WIN(win, info_args).same_disp_unit = 0;
+        }
+        else if (!strcmp(curr_ptr->key, "alloc_shared_noncontig")) {
+            if (!strcmp(curr_ptr->value, "true"))
+                MPIDI_CH4U_WIN(win, info_args).alloc_shared_noncontig = 1;
+            else
+                MPIDI_CH4U_WIN(win, info_args).alloc_shared_noncontig = 0;
+        }
+      next:
```

Comment:
I think this logic is somewhat dangerous. What if a user supplies a bogus value like "foobar"?